esprima.js

```diff
@@ -2104,10 +2104,7 @@
         finishProgram: function (body) {
             this.type = Syntax.Program;
             this.body = body;
-            if (sourceType === 'module') {
-                // very restrictive for now
-                this.sourceType = sourceType;
-            }
+            this.sourceType = sourceType;
```

Comment:
Pass the `sourceType` from the function parameter as all the AST node constructor is not supposed to rely much on global(s).